crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -508,20 +504,37 @@ where
         trace!(target: "consensus::engine", ?state, "Received new forkchoice state update");
 
         match self.pre_validate_forkchoice_update(state) {
-            Ok(Some(on_updated)) => {
-                // Pre-validate forkchoice state update and return if it's invalid
-                // or cannot be processed at the moment.
-                self.on_forkchoice_updated_status(state, on_updated, tx);
-            }
-            Ok(None) => {
-                self.set_blockchain_tree_action(
-                    BlockchainTreeAction::MakeForkchoiceHeadCanonical { state, attrs, tx },
-                );
+            Ok(on_updated_result) => {
+                if let Some(on_updated) = on_updated_result {
+                    // Pre-validate forkchoice state update and return if it's invalid
+                    // or cannot be processed at the moment.
+                    self.on_forkchoice_updated_status(state, on_updated, tx);
+                } else if let Some(hook) = self.hooks.active_db_write_hook() {
```

Comment:
why not match arms?

Reply:
no particular reason, there would still be another `if let Some else` in `Ok(None)` arm